test/e2e/flask/multichain-api/evm/wallet_invokeMethod.spec.ts

```diff
@@ -175,48 +180,66 @@ describe('Multichain API', function () {
+              network: string;
+            }[] = [];
+            for (const [i, scope] of GANACHE_SCOPES.entries()) {
+              expectedConfirmations.push({
+                account:
+                  i === INDEX_FOR_ALTERNATE_ACCOUNT ? 'Account 2' : 'Account 1',
+                network: SCOPE_TO_NETWORK_NAME[scope],
+              });
+            }
 
-            // check which account confirmation page is displayed on second screen
+            const resultConfirmations: { account: string; network: string }[] =
+              [];
             await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
-            await confirmation.checkPageIsLoaded();
-            const screenForAccount2 =
-              await confirmation.checkIsSenderAccountDisplayed('Account 2');
-            if (screenForAccount2) {
-              await confirmation.checkNetworkIsDisplayed('Localhost 8546');
-              await confirmation.clickFooterConfirmButton();
-
-              // third confirmation page should display Account 1 as sender account
-              await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
-              await confirmation.checkPageIsLoaded();
-              assert.equal(
-                await confirmation.checkIsSenderAccountDisplayed('Account 1'),
-                true,
-              );
-              await confirmation.checkNetworkIsDisplayed('Localhost 7777');
-              await confirmation.clickFooterConfirmButtonAndAndWaitForWindowToClose();
-            } else {
-              await confirmation.checkNetworkIsDisplayed('Localhost 7777');
-              await confirmation.clickFooterConfirmButton();
+            const firstConfirmation = new TransactionConfirmation(driver);
+            await firstConfirmation.checkPageIsLoaded();
+            let currentAccount = await firstConfirmation.getSenderAccountName();
+            let currentNetwork = await firstConfirmation.getNetworkName();
+            await firstConfirmation.clickFooterConfirmButton();
+            resultConfirmations.push({
+              account: currentAccount,
+              network: currentNetwork,
+            });
 
-              // third confirmation page should display Account 2 as sender account
+            // Collect actual confirmations from each confirmation screen
+            for (let i = 0; i < GANACHE_SCOPES.length - 1; i++) {
               await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+              const confirmation = new TransactionConfirmation(driver);
               await confirmation.checkPageIsLoaded();
-              assert.equal(
-                await confirmation.checkIsSenderAccountDisplayed('Account 2'),
-                true,
-              );
-              await confirmation.checkNetworkIsDisplayed('Localhost 8546');
-              await confirmation.clickFooterConfirmButtonAndAndWaitForWindowToClose();
+              await confirmation.checkNetworkIsNotDisplayed(currentNetwork);
```

Comment:
in each iteration, we check that the network is different from the previous confirmation screen to ensure that we’ve switched to a new confirmation screen